Sources/OpenAPIRuntime/Base/Acceptable.swift

```diff
@@ -0,0 +1,164 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// The protocol that all generated `AcceptableContentType` enums conform to.
+public protocol AcceptableProtocol: RawRepresentable, Sendable, Equatable, Hashable, CaseIterable
+where RawValue == String {}
+
+/// A quality value used to describe the order of priority in a comma-separated
+/// list of values, such as in the Accept header.
+public struct QualityValue: Sendable, Equatable, Hashable {
+
+    /// As the quality value only retains up to and including 3 decimal digits,
+    /// we store it in terms of the thousands.
+    ///
+    /// This allows predictable equality comparisons and sorting.
+    ///
+    /// For example, 1000 thousands is the quality value of 1.0.
+    private let thousands: UInt16
+
+    /// Creates a new quality value of the default value 1.0.
+    public init() {
+        self.thousands = 1000
+    }
+
+    /// Returns a Boolean value indicating whether the quality value is
+    /// at its default value 1.0.
+    public var isDefault: Bool {
+        thousands == 1000
+    }
+
+    /// Creates a new quality value from the provided floating-point number.
+    ///
+    /// - Precondition: The value must be between 0.0 and 1.0, inclusive.
+    public init(doubleValue: Double) {
+        precondition(
+            doubleValue >= 0.0 && doubleValue <= 1.0,
+            "Input number into quality value is out of range"
+        )
+        self.thousands = UInt16(doubleValue * 1000)
+    }
+
+    /// The value represented as a floating-point number between 0.0 and 1.0, inclusive.
+    public var doubleValue: Double {
+        Double(thousands) / 1000
+    }
+}
+
+extension QualityValue: RawRepresentable {
+    public init?(rawValue: String) {
+        guard let doubleValue = Double(rawValue) else {
+            return nil
+        }
+        self.init(doubleValue: doubleValue)
+    }
+
+    public var rawValue: String {
+        String(format: "%0.3f", doubleValue)
+    }
+}
+
+extension QualityValue: ExpressibleByIntegerLiteral {
+    public init(integerLiteral value: UInt16) {
+        self.thousands = value * 1000
```

Comment:
Feels like we should also precondition here that we're within a valid range - and multiplying by 1000 is probably unwanted too.

Reply:
Good catch - added a precondition.

Multiplying it is correct, as the internal storage is in thousands, so 1000 maps to quality value 1.

Reply:
https://github.com/apple/swift-openapi-runtime/pull/37/commits/15d2b021ec6f88c5d845a3d59371311ac44de2cd

Reply:
Ah okay, I kinda assumed that if using this init, we would be passing a `thousands` value and not actually a quality value, although I understand that's confusing and wrong. So essentially the only allowed values here are 0 or 1, right?

Reply:
Yup.